src/groupederror.jl

```diff
@@ -0,0 +1,137 @@
+type GroupedError
+    x
+    y
+    err
+    group
+    xaxis
+end
+
+get_axis(column::PooledDataArray) = sort!(unique(column))
+get_axis(column::AbstractArray) = linspace(minimum(column),maximum(column),200)
+
+# f is the function used to analyze dataset: define it as nan when it is not defined,
+# the input is: dataframe used, points chosen on the x axis, x (and maybe y) column labels
+# the output is the y value for the given xvalues
+
+function get_mean_sem(trend,variation, f, df, population, args...; xaxis = false, kwargs...)
+    # define points on x axis
+    if xaxis == false
+        xvalues = get_axis(df[args[1]])
+    else
+        xvalues = xaxis
+    end
+
+    if population == []
+        mean_across_pop = f(df,xvalues, args...; kwargs...)
+        sem_across_pop = zeros(length(xvalues));
+        valid = ~isnan(mean_across_pop)
+    else
+        # get mean value and sem of function of interest
+        splitdata = groupby(df, population)
+        v = Array(Float64, length(xvalues), length(splitdata));
```

Comment:
It would be cool to define this mean_and_sem behaviour in a separate function tha tworks on a `GroupedDataFrame` (the type of splitdata), has the same argument structure as the full function (minus `population`), and is called here. That could be exported as well, then a user could call this directly on a `DataFrame` that he had already split himself with `groupby`.